geode-experimental-driver/src/main/java/org/apache/geode/experimental/driver/ValueEncoder.java

```diff
@@ -47,15 +47,10 @@ public ValueEncoder(ValueSerializer valueSerializer) {
   BasicTypes.EncodedValue encodeValue(Object unencodedValue) {
     BasicTypes.EncodedValue.Builder builder = BasicTypes.EncodedValue.newBuilder();
 
-    ByteString customBytes;
-    try {
-      customBytes = valueSerializer.serialize(unencodedValue);
-    } catch (IOException e) {
-      throw new IllegalStateException(e);
-    }
-
-    if (customBytes != null) {
-      return builder.setCustomObjectResult(customBytes).build();
+    if (valueSerializer.supportsPrimitives()) {
+      ByteString customBytes = customSerialize(unencodedValue);
+      if (customBytes != null)
```

Comment:
If the value serializer returns true for supportsPrimitives(), should it still be allowed to return null here?